x-pack/test/apm_api_integration/tests/cold_start/cold_start.spec.ts

```diff
@@ -0,0 +1,174 @@
+          expect(body.currentPeriod.transactionColdstartRate.every(({ y }) => y === 0.5)).to.be(
+            true
+          );
+        });
+
+        it('returns correct average rate', () => {
+          expect(body.currentPeriod.average).to.be(0.5);
+        });
+
+        it("doesn't have data for the previous period", () => {
+          expect(body).to.have.property('previousPeriod');
+          expect(body.previousPeriod.transactionColdstartRate).to.have.length(0);
+          expect(body.previousPeriod.average).to.be(null);
+        });
+      });
+
+      describe('with comparison', () => {
+        let body: ColdStartRate;
+        let status: number;
+
+        before(async () => {
+          await generateData({ synthtraceEsClient, start, end });
+
+          const fiveMinutes = 5 * 60 * 1000;
+          const response = await callApi({
+            query: {
+              start: new Date(end - fiveMinutes).toISOString(),
+              end: new Date(end).toISOString(),
+              comparisonStart: new Date(start).toISOString(),
+              comparisonEnd: new Date(start + fiveMinutes).toISOString(),
+            },
+          });
+          body = response.body;
+          status = response.status;
+        });
+
+        after(() => synthtraceEsClient.clean());
+
+        it('returns correct HTTP status', () => {
+          expect(status).to.be(200);
+        });
+
+        it('returns some data', () => {
+          expect(body.currentPeriod.average).not.to.be(null);
+          expect(body.currentPeriod.transactionColdstartRate.length).to.be.greaterThan(0);
+          const hasCurrentPeriodData = body.currentPeriod.transactionColdstartRate.some(({ y }) =>
+            isFiniteNumber(y)
+          );
+          expect(hasCurrentPeriodData).to.equal(true);
+
+          expect(body.previousPeriod.average).not.to.be(null);
+          expect(body.previousPeriod.transactionColdstartRate.length).to.be.greaterThan(0);
+          const hasPreviousPeriodData = body.previousPeriod.transactionColdstartRate.some(({ y }) =>
+            isFiniteNumber(y)
+          );
+          expect(hasPreviousPeriodData).to.equal(true);
+        });
+
+        it('has same start time for both periods', () => {
+          expect(first(body.currentPeriod.transactionColdstartRate)?.x).to.equal(
+            first(body.previousPeriod.transactionColdstartRate)?.x
+          );
+        });
+
+        it('has same end time for both periods', () => {
+          expect(last(body.currentPeriod.transactionColdstartRate)?.x).to.equal(
+            last(body.previousPeriod.transactionColdstartRate)?.x
+          );
+        });
+
+        it('returns same number of buckets for both periods', () => {
+          expect(body.currentPeriod.transactionColdstartRate.length).to.be(
+            body.previousPeriod.transactionColdstartRate.length
+          );
+        });
+
+        it('has same average value for both periods', () => {
```

Comment:
ideally we index different values (and test for them as well)

Reply:
Added different values and now testing for all of them